icepyx/core/types.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import annotations
+
+from typing import Literal, TypedDict
+from typing_extensions import NotRequired
+
+
+CMRParamsBase = TypedDict(
+    "CMRParamsBase",
+    {
+        "temporal": NotRequired[str],
+        "options[readable_granule_name][pattern]": NotRequired[str],
+        "options[spatial][or]": NotRequired[str],
+        "readable_granule_name[]": NotRequired[str],
+    },
+)
+
+
+class CMRParamsWithBbox(CMRParamsBase):
+    bounding_box: str
+
+
+class CMRParamsWithPolygon(CMRParamsBase):
+    polygon: str
+
+
+CMRParams = CMRParamsWithBbox | CMRParamsWithPolygon
+
+
+class EGISpecificParamsBase(TypedDict):
```

Comment:
I'm like sure I've got stuff wrong here, but it's right enough to pass static analysis :laughing:

Reply:
It looks like we don't have any parameters set up for subsetting. This especially comes into play because EGI takes different parameters than CMR for time and space (`time` vs `temporal` and `bbox` versus `bounding_box`).

Reply:
df78ca2